RefrigeratorSaraFarkash/RefrigeratorSaraFarkash/CRefrigerator.cs

```diff
@@ -0,0 +1,334 @@
+                        itemsToRemove.Add(item);
+                    }
+                }
+                if (itemsToRemove.Count >= 0)
+                {
+                    // Since it is impossible to delete while in a loop, I created a list that stores all the expired products.
+                    foreach (CItem item in itemsToRemove)
+                    {
+                        CItem itemremoving = removeItemFromRef(item.Id);
+                    }
+                }
+            }
+        }
+
+        //6
+        public List<CItem> wantEat(string kashroot, string typeeat)
+        {
+            List<CItem> wanttoeat = new List<CItem>();
+
+            for (int i = 0; i < shelfs.Count; i++)
+            {
+                foreach (CItem item in shelfs[i].Items)
+                {
+                    if (item.Kashroot == kashroot && item.TypeI == typeeat && item.ExpiryDate > DateTime.Today)
+                        wanttoeat.Add(item);
+                }
+            }
+            return wanttoeat;
+        }
+
+        // sort
+
+        public int CompareTo(CRefrigerator otherRefrigerator)
+        {
+            return ((this.spaceLeftInRefrigerator() > otherRefrigerator.spaceLeftInRefrigerator()) ? (-1) : (this.spaceLeftInRefrigerator() == otherRefrigerator.spaceLeftInRefrigerator()) ? 0 : 1);
+        }
+
+        public List<CItem> SortItems()
+        {
+            List<CItem> items = new List<CItem>();
+            foreach (CShelf shelf in this.shelfs)
+            {
+                items.AddRange(shelf.Items);
+            }
+            items.Sort();
+            return items;
+        }
+        public List<CShelf> SortShelf()
+        {
+            List<CShelf> shelfs = new List<CShelf>();
+            foreach (CShelf shelf in this.shelfs)
+            {
+                shelfs.Add(shelf);
+            }
+            shelfs.Sort();
+
+            return shelfs;
+        }
+        //1
+        //
+        //A function that receives a date and returns an list with all the food deleted up to the date
+        public List<CItem> wantToREMOVE(int untilday, string KashrootItem)
+        {
+            List<CItem> RemoveItem = new List<CItem>();
+            DateTime lastDate = DateTime.Today;
+            lastDate = lastDate.AddDays(untilday);
+            for (int i = 0; i < this.numberOfShelves; i++)
+            {
+                foreach (CItem item in this.Shelfs[i].Items)
+                {
+                    if (item.ExpiryDate <= lastDate && KashrootItem == item.Kashroot)
+                    {
+                        RemoveItem.Add(item);
+                    }
+                }
+                foreach (CItem itemToRemove in RemoveItem)
+                {
+                    CItem itemremving = removeItemFromRef(itemToRemove.Id);
+                }
+            }
+            return RemoveItem;
+        }
+        public Boolean ThereIsPlacefor20()
```

Comment:
There is no point to this method.
Although you do this calculations several times it is a one liner....